package/MDAnalysis/analysis/msd.py

```diff
@@ -421,3 +456,46 @@ def _conclude_fft(self):  # with FFT, np.float64 bit prescision required.
                 positions[:, n, :]
             )
         self.results.timeseries = self.results.msds_by_particle.mean(axis=1)
+    
+
+    def _conclude_modified(self):
+        from collections import defaultdict
+        print("The Dump file has non-linear time spacing")
+        
+        dump_times = [ts.time for ts in self._trajectory]
```

Comment:
This iterates through the whole trajectory and may already be expensive!

Maybe unavoidable but keep it in mind.

Reply:
You would need to use `self._sliced_trajectory` instead of `self._trajectory`. If someone is running it with `run(start=10, stop=1000, step=5)` then `self._sliced_trajectory` will contain the proper frames that were analyzed in `_single_frame`. However, the time information should already by available in [self.times](https://docs.mdanalysis.org/stable/documentation_pages/analysis/base.html#MDAnalysis.analysis.base.AnalysisBase.times):
```suggestion
        dump_times = self.times
```
This should improve performance as you don't have to iterate through the trajectory again.

Reply:
And it's already a numpy array.